tutorials/migration.md

```diff
@@ -4,96 +4,846 @@ layout: intro
-- It is possible to provide your own custom icons in the `conf/icons/classic` folder - no need to overwrite the icons that come with the runtime.
-- For custom dynamic icons depending on an item state, you need in openHAB 2 to define in the `conf/icons/classic` folder a default icon (without the state part) in addition to the icons for the different states. This default icon was not required in openHAB 1. For example, for a Wifi level that can take a value from 1 to 4, in addition to the files wifi-1.png, wifi-2.png, wifi-3.png and wifi-4.png, you need to provide a file named wifi.png.
-- Note that Basic UI requires icons with SVG format by default; a Basic UI setting is available to choose between SVG and PNG formats.
+[2. Backups](#backups)
 
-### Rules
+[3. Shutdown openHAB 1.x](#shutdown-openhab-1x)
 
-In order to continue using rules from openHAB 1, a few minor changes might be necessary:
+[4. Install openHAB 2](#install-openhab-2)
 
-1. Import statements at the top are not required anymore for any org.openhab package, so these can be removed
-1. The state "Uninitialized" has been renamed to "NULL" and thus needs to be replaced in the rules
-1. The `HSBType` state type can no longer be constructed using a `java.awt.Color` object, and there is no longer a `toColor()` method.  Use the following alternatives:
+[5. Prepare openHAB 2 for the Existing openHAB 1.x Configuration](#prepare-openhab-2-for-the-existing-openhab-1x-configuration)
 
-```java
+&nbsp;&nbsp;[5.1 Text Based](#text-based)
+
+&nbsp;&nbsp;[5.2 Karaf Console](#karaf-console)
+
+&nbsp;&nbsp;[5.3 PaperUI Approach](#paperui-approach)
+
+&nbsp;&nbsp;[5.4 Configure Add-ons](#configure-add-ons)
+
+&nbsp;&nbsp;[5.5 Installing Unofficially Supported openHAB 1.x Add-ons](#installing-unofficially-supported-openhab-1x-add-ons)
+
+&nbsp;&nbsp;[5.6 Final Add-ons Installation Steps](#installing-unofficially-supported-openhab-1x-add-ons)
+
+[6. Necessary Changes](#necessary-changes)
+
+&nbsp;&nbsp;[6.1 Items](#items)
+
+&nbsp;&nbsp;[6.2 Sitemap](#sitemap)
+
+&nbsp;&nbsp;[6.3 Rules](#rules)
+
+[7. Testing](#testing)
+
+[8. New Concepts: Things and Channels](#new-concepts-things-and-channels)
+
+[9. Retire openHAB 1.x](#retire-openhab-1x)
+
+[10. Migrating to openHAB 2 Bindings](#migrating-to-openhab-2-bindings)
+
+&nbsp;&nbsp;[10.1 Eclipse SmartHome Designer](#eclipse-smarthome-designer)
+
+&nbsp;&nbsp;[10.2 My.openHAB](#myopenhab)
+
+&nbsp;&nbsp;[10.3 Other Bindings](#other-bindings)
+
+[11. Managing the Inbox, Things, and Channels](#managing-the-inbox-things-and-channels)
+
+&nbsp;&nbsp;[11.1 Managing the Inbox using PaperUI](#managing-the-inbox-using-paperui)
+
+&nbsp;&nbsp;[11.2 Managing the Inbox Using the Karaf Console](#managing-the-inbox-using-the-karaf-console)
+
+&nbsp;&nbsp;[11.3 Linking Channels to Items](#managing-the-inbox-using-the-karaf-console)
+
+&nbsp;&nbsp;[11.4 Manually Creating Things](#manually-creating-things)
+
+##Preparation
```

Comment:
There needs to be spaces between # and the section title, otherwise jekyll won't render it. Multiple instances of this throughout this PR.